### PR TITLE
Fix heuristics for Talk namespace appending and handling.

### PR DESCRIPTION
**Pull request by dbrant**
https://phabricator.wikimedia.org/T280437

### CODE REVIEW

File: app/src/main/java/org/wikipedia/page/PageTitle.java [hunk 1]
```diff
@@ -150,15 +150,20 @@ public PageTitle(@Nullable String text, @NonNull WikiSite wiki, @Nullable String
             if (Arrays.asList(Locale.getISOLanguages()).contains(namespaceOrLanguage)) {
                 this.namespace = null;
                 this.wiki = new WikiSite(wiki.authority(), namespaceOrLanguage);
-            } else {
+                this.text = TextUtils.join(":", Arrays.copyOfRange(parts, 1, parts.length));
+            } else if (parts[1].length() > 0 && !Character.isWhitespace(parts[1].charAt(0)) && (parts[1].charAt(0) != '_')) {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/page/PageTitle.java:154]: Really minor but the parentheses of  `(parts[1].charAt(0) != '_')` can be removed.

Comment (dbrant) [reply app/src/main/java/org/wikipedia/page/PageTitle.java]: done

**Review by cooltey — APPROVED** [review]
Looks good to me. @sharvaniharan Please check and merge if looks good to you too.

**Comment by cooltey** [comment]
I'll just merge it and please leave a comment if you see something else @sharvaniharan.

**Pull request merged**